app/controllers/my_facilities_controller.rb

```diff
@@ -113,4 +89,17 @@ def force_cache?
   def report_with_exclusions?
     current_admin.feature_enabled?(:report_with_exclusions)
   end
+
+  def process_facility_stats(type)
+    display_facilities = filter_facilities
+    @data_for_facility = {}
+    data_for_stats = {}
+    @accessible_facilities.each do |facility|
+      result = Reports::RegionService.new(region: facility, period: @period).call
+      @data_for_facility[facility.name] = result if display_facilities.include?(facility)
+      data_for_stats[facility.name] = result
```

Comment:
Okay, now I understand the change here wrt filtering and such.

This is going to add a **substantial** amount of processing time. Consider:

#### Before
* we find what facilities a user has `view_reports` access to, apply the filters from the UI (for size, district, etc), and then grab numbers for that subset via RegionService

#### After
* we find what facilities a user has `view_reports` access to, grab numbers for **all** of them via RegionService, and then apply filters for things we want to display in the view

So for a power user, this could be going from ~100/200 calls to RegionService to well over a thousand. I would expect this to be a problem with a production-sized dataset.

Reply:
Yeah, this change was requested late last week. I may have made a bad assumption about what was being queried when you go to http://localhost:3000/my_facilities/bp_controlled. I looked at how filtered facilities were being set, but I should have done some manual testing and asked more questions. The default filtered facilities uses some fallbacks, like using the first facility group instead of all facility groups, which significantly decreases the scope. I'll talk to Claudio about which facilities we really care about. It may still be all accessible and you may be right that it could be a problem. Maybe we only care about all facilities within the current group.

Reply:
> Maybe we only care about all facilities within the current group.

👍 Yeah that sounds like it a more reasonable group to grab the data for.

Any page that grabs report data for _all_ facilities is going to have a bad time. We don't have good summary tables or caching strategies in place for that sort of thing (yet).